### PR TITLE
PV61 sync

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by gavofyork** [comment]
all failing?

File: libp2p/Session.cpp [hunk 1]
```diff
@@ -386,11 +386,12 @@ void Session::drop(DisconnectReason _reason)
 void Session::disconnect(DisconnectReason _reason)
 {
 	clog(NetConnect) << "Disconnecting (our reason:" << reasonOf(_reason) << ")";
+	size_t peerCount = m_server->peerCount(); //needs to  be outside of lock to avoid deadlocking with other thread that capture x_info/x_sessions in reverse order
 	DEV_GUARDED(x_info)
 		StructuredLogger::p2pDisconnected(
 			m_info.id.abridged(),
 			m_peer->endpoint, // TODO: may not be 100% accurate
-			m_server->peerCount()
+			peerCount
```
Comment (gavofyork) [inline libp2p/Session.cpp]: ??

**Comment by arkpar** [comment]
waiting on #2336 to rebase

File: libp2p/Session.cpp [hunk 2]
```diff
@@ -315,7 +315,7 @@ void Session::drop(DisconnectReason _reason)
 	if (socket.is_open())
 		try
 		{
-			clog(NetConnect) << "Closing " << socket.remote_endpoint() << "(" << reasonOf(_reason) << ")";
+			clog(NetWarn) << "Closing " << socket.remote_endpoint() << "(" << reasonOf(_reason) << ")";
```
Comment (gavofyork) [inline libp2p/Session.cpp]: reason? this happens all the time with geth nodes.

Comment (arkpar) [inline libp2p/Session.cpp]: debugging artefact, reverted

**Pull request merged**